### ISSUE TITLE
TNS name doesn't work

### ISSUE THREAD

**Issue by LukashonakV** [jenkinsci/sqlplus-script-runner-plugin#43]
Dear maintainer, I've faced with the issue with the sqlplus-script-runner-plugin.

Ubuntu 18.04 bionic
Jenkins ver. 2.190.1
x86_64 Linux 4.15.0-65-generic
SQL*Plus: Release 12.1.0.2.0 Production

I've got defined TNS_NAMES and filled "Set global TNS_ADMIN" by it.
When I'm trying to start project, it fails with the error ORA-12154: TNS:could not resolve the connect identifier specified.

In the same time I copy connection string from the console to SQLPLUS and it's connected as well(what is expected due to I've got appropriate connection string in the tnsnames.ora).

The only method which works well on my machine with Jenkins when I point the whole connection string like (db.com:1523/TEST_SID). Last string means that plugin defined TNS NAMES path as well... but for some reason doesn't recognize connection string from tnsnames.ora

**Comment by boaglio** [comment]
Please turn on debug option and post here a full log of your job with error.

**Comment by LukashonakV** [comment]
```
Started by user admin
Running as SYSTEM
Building in workspace /var/lib/jenkins/workspace/Demo
--------------------------------------------------------------------------
Getting ORACLE_HOME...
[DEBUG] Global TNS_ADMIN selected
--------------------------------------------------------------------------
Using global TNS_ADMIN location
[DEBUG] Try to detect ORACLE_HOME
--------------------------------------------------------------------------
Using detected ORACLE_HOME
 detected host = localhost
 slave machine ? false
--------------------------------------------------------------------------
Using ORACLE_HOME =/usr/lib/oracle/12.1/client64
--------------------------------------------------------------------------
Getting SQLPlus version
[DEBUG] ORACLE_HOME = /usr/lib/oracle/12.1/client64
[DEBUG] LD_LIBRARY_PATH = /usr/lib/oracle/12.1/client64/lib
SQL*Plus exec file = sqlplus
found SQL*Plus on /usr/lib/oracle/12.1/client64/bin
--------------------------------------------------------------------------
[DEBUG] Statement:
/usr/lib/oracle/12.1/client64/bin/sqlplus -v  
--------------------------------------------------------------------------
[Demo] $ /usr/lib/oracle/12.1/client64/bin/sqlplus -v

SQL*Plus: Release 12.1.0.2.0 Production

Process exited with status 0
--------------------------------------------------------------------------
 detected host = localhost
[DEBUG] testing directory /usr/lib/oracle/12.1/client64
--------------------------------------------------------------------------
Using ORACLE_HOME =/usr/lib/oracle/12.1/client64
--------------------------------------------------------------------------
Running defined script on DB_DEMO/********@SID_DEMO
Temp script: file:/tmp/temp-script-15701028134146979309550885701231.sql
--------------------------------------------------------------------------
[DEBUG] ORACLE_HOME = /usr/lib/oracle/12.1/client64
[DEBUG] LD_LIBRARY_PATH = /usr/lib/oracle/12.1/client64/lib:/usr/lib/oracle/12.1/client64
Master Work Directory = /var/lib/jenkins/workspace/Demo
SQL*Plus exec file = sqlplus
[DEBUG] found SQL*Plus on /usr/lib/oracle/12.1/client64/bin
[DEBUG]  Statement: 
--------------------------------------------------------------------------
/usr/lib/oracle/12.1/client64/bin/sqlplus -L DB_DEMO/DB_DEMO@SID_DEMO @/tmp/temp-script-15701028134146979309550885701231.sql  
--------------------------------------------------------------------------
 
[Demo] $ /usr/lib/oracle/12.1/client64/bin/sqlplus -L DB_DEMO/DB_DEMO@SID_DEMO @/tmp/temp-script-15701028134146979309550885701231.sql

SQL*Plus: Release 12.1.0.2.0 Production on Thu Oct 3 14:40:13 2019

Copyright (c) 1982, 2014, Oracle.  All rights reserved.

ERROR:
ORA-12154: TNS:could not resolve the connect identifier specified


SP2-0751: Unable to connect to Oracle.  Exiting SQL*Plus
Process exited with status 1
--------------------------------------------------------------------------
Exit code: 1
--------------------------------------------------------------------------
java.lang.RuntimeException: Process exited with error
	at org.jenkinsci.plugins.sqlplus.script.runner.SQLPlusRunner.run(SQLPlusRunner.java:444)
	at org.jenkinsci.plugins.sqlplus.script.runner.SQLPlusRunnerBuilder.perform(SQLPlusRunnerBuilder.java:174)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:79)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:741)
	at hudson.model.Build$BuildExecution.build(Build.java:206)
	at hudson.model.Build$BuildExecution.doRun(Build.java:163)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:504)
	at hudson.model.Run.execute(Run.java:1815)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:429)
ERROR: Process exited with error
Finished: FAILURE
```

In the same time I just copy connection string from this log to bash /usr/lib/oracle/12.1/client64/bin/sqlplus -L DB_DEMO/DB_DEMO@SID_DEMO and it's OK.

**Comment by boaglio** [comment]
Did you set TNS_ADMIN to where tnsnames.ora is ? What is the value of  your TNS_ADMIN ?

**Comment by LukashonakV** [comment]
The value for TNS_ADMIN is : TNS_ADMIN=/home/viktar/Дакументы/Oracle. This is the place where tnsnames.ora is situdated. For at first I thought this issue might be come with the RU locale in the path of the variable. And I changed TNS_ADMIN path where used the only english alphabet(of course didn't forget to move tnsnames.ora). For to be sure I checked it with SQLPlus. It worked fine. But Jenkins plugin unfortunatelly still have this issue. Anyway if TNS_ADMIN has got wrong value I wouldn't be able to use it in SQLPlus in connection string.

**Comment by boaglio** [comment]
I updated this release and added a feature to try to find _tnsnames.ora_ file in your custom / global  TNS_ADMIN directory. Please update and retry.

**Comment by LukashonakV** [comment]
Unfortunately no luck.

```
12:31:28 Started by user admin
12:31:28 Running as SYSTEM
12:31:28 Building in workspace /var/lib/jenkins/workspace/Demo
12:31:28 --------------------------------------------------------------------------
12:31:28 Getting ORACLE_HOME...
12:31:28 [DEBUG] Global TNS_ADMIN selected
12:31:28 --------------------------------------------------------------------------
12:31:28 Using global TNS_ADMIN location
12:31:28 TNS_ADMIN >>> /home/viktar/Дакументы/Oracle
12:31:28 [DEBUG] Try to detect ORACLE_HOME
12:31:28 --------------------------------------------------------------------------
12:31:28 Using detected ORACLE_HOME
12:31:28  detected host = localhost
12:31:28  slave machine ? false
12:31:28 --------------------------------------------------------------------------
12:31:28 Using ORACLE_HOME =/usr/lib/oracle/12.1/client64
12:31:28 --------------------------------------------------------------------------
12:31:28 Getting SQLPlus version
12:31:28 [DEBUG] ORACLE_HOME = /usr/lib/oracle/12.1/client64
12:31:28 [DEBUG] LD_LIBRARY_PATH = /usr/lib/oracle/12.1/client64/lib
12:31:28 SQL*Plus exec file = sqlplus
12:31:28 found SQL*Plus on /usr/lib/oracle/12.1/client64/bin
12:31:28 --------------------------------------------------------------------------
12:31:28 [DEBUG] Statement:
12:31:28 /usr/lib/oracle/12.1/client64/bin/sqlplus -v  
12:31:28 --------------------------------------------------------------------------
12:31:28 [Demo] $ /usr/lib/oracle/12.1/client64/bin/sqlplus -v
12:31:28 
12:31:28 SQL*Plus: Release 12.1.0.2.0 Production
12:31:28 
12:31:28 Process exited with status 0
12:31:28 --------------------------------------------------------------------------
12:31:28  detected host = localhost
12:31:28 [DEBUG] testing directory /usr/lib/oracle/12.1/client64
12:31:28 --------------------------------------------------------------------------
12:31:28 Using ORACLE_HOME =/usr/lib/oracle/12.1/client64
12:31:28 --------------------------------------------------------------------------
12:31:28 Running defined script on DB_DEMO/********@SID_DEMO
12:31:28 Temp script: file:/tmp/temp-script-15706998885592099134206016593643.sql
12:31:28 --------------------------------------------------------------------------
12:31:28 [DEBUG] ORACLE_HOME = /usr/lib/oracle/12.1/client64
12:31:28 [DEBUG] LD_LIBRARY_PATH = /usr/lib/oracle/12.1/client64/lib:/usr/lib/oracle/12.1/client64
12:31:28 [DEBUG] found TNSNAMES.ORA on /home/viktar/Дакументы/Oracle
12:31:28 Master Work Directory = /var/lib/jenkins/workspace/Demo
12:31:28 SQL*Plus exec file = sqlplus
12:31:28 [DEBUG] found SQL*Plus on /usr/lib/oracle/12.1/client64/bin
12:31:28 [DEBUG]  Statement: 
12:31:28 --------------------------------------------------------------------------
12:31:28 /usr/lib/oracle/12.1/client64/bin/sqlplus -L DB_DEMO/DB_DEMO@SID_DEMO @/tmp/temp-script-15706998885592099134206016593643.sql  
12:31:28 --------------------------------------------------------------------------
12:31:28  
12:31:28 [Demo] $ /usr/lib/oracle/12.1/client64/bin/sqlplus -L DB_DEMO/DB_DEMO@SID_DEMO @/tmp/temp-script-15706998885592099134206016593643.sql
12:31:28 
12:31:28 SQL*Plus: Release 12.1.0.2.0 Production on Thu Oct 10 12:31:28 2019
12:31:28 
12:31:28 Copyright (c) 1982, 2014, Oracle.  All rights reserved.
12:31:28 
12:31:28 ERROR:
12:31:28 ORA-12154: TNS:could not resolve the connect identifier specified
12:31:28 
12:31:28 
12:31:28 SP2-0751: Unable to connect to Oracle.  Exiting SQL*Plus
12:31:28 Process exited with status 1
12:31:28 --------------------------------------------------------------------------
12:31:28 Exit code: 1
12:31:28 --------------------------------------------------------------------------
12:31:28 java.lang.RuntimeException: Process exited with error
12:31:28 	at org.jenkinsci.plugins.sqlplus.script.runner.SQLPlusRunner.run(SQLPlusRunner.java:448)
12:31:28 	at org.jenkinsci.plugins.sqlplus.script.runner.SQLPlusRunnerBuilder.perform(SQLPlusRunnerBuilder.java:174)
12:31:28 	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:79)
12:31:28 	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
12:31:28 	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:741)
12:31:28 	at hudson.model.Build$BuildExecution.build(Build.java:206)
12:31:28 	at hudson.model.Build$BuildExecution.doRun(Build.java:163)
12:31:28 	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:504)
12:31:28 	at hudson.model.Run.execute(Run.java:1815)
12:31:28 	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
12:31:28 	at hudson.model.ResourceController.execute(ResourceController.java:97)
12:31:28 	at hudson.model.Executor.run(Executor.java:429)
12:31:28 ERROR: Process exited with error
12:31:28 Finished: FAILURE
```

In the same time echo from bash

```
viktar@Barakuda:~$ /usr/lib/oracle/12.1/client64/bin/sqlplus -L DB_DEMO/DB_DEMO@SID_DEMO

SQL*Plus: Release 12.1.0.2.0 Production on Thu Oct 10 12:37:07 2019

Copyright (c) 1982, 2014, Oracle.  All rights reserved.


Connected to:
Oracle Database 11g Release 11.2.0.4.0 - 64bit Production

SQL> 
```

**Comment by boaglio** [comment]
Maybe there is something wrong in _Jenkins_ user environment (where Jenkins app runs) and not with _viktar_ user. 
You can try to do this workaround :

```
sudo su - jenkins
mkdir /usr/lib/oracle/12.1/client64/network
ln -s /home/viktar/Дакументы/Oracle /usr/lib/oracle/12.1/client64/network/admin 
```

**Comment by LukashonakV** [comment]
Thank you for your patience.
1 . Oracle instaclient 12.X for Linux for some reason haven't brought network folder in release. I just installed the freshest one (19.3) and folders strucuture now is looking good.
2 . I've just realized Jenkins is using it's own user (jenkins). And eventually the only user wich is able even to read tnsnames.ora was my own. In the same time according TNS_ADMIN system variable all users must have an opportunity to read it. So I just allowed to read tnsnames.ora to everyone and now Jenkins and this plugin works well. 

Thank you a lot :)

**Comment by karthikeyan82** [comment]
I am having the same issue. I set permissions but still not working

I confirmed TNS_admin has the TNSNAMES.ORA file

Getting ORACLE_HOME...
Using custom SQL*Plus location
SQL*Plus directory: /bin/sqlplus
Using custom TNS_ADMIN location
TNS_ADMIN: /apps/jenkins-agent/workspace/HP-BIO-SNAP/DeployScripts
--------------------------------------------------------------------------
Using custom ORACLE_HOME
--------------------------------------------------------------------------
Using ORACLE_HOME =/usr/lib/oracle/21/client64
--------------------------------------------------------------------------
Getting SQLPlus version
[DeployScripts] $ /bin/sqlplus -v

SQL*Plus: Release 21.0.0.0.0 - Production
Version 21.1.0.0.0

Process exited with status 0
--------------------------------------------------------------------------
--------------------------------------------------------------------------
Using ORACLE_HOME =/usr/lib/oracle/21/client64
--------------------------------------------------------------------------
Running defined script on csar_ADM/********@csard02r.pldc.kp.org
Temp script: file:/apps/jenkins-agent/workspace/HP-BIO-SNAP/DeployScripts/temp-script-16167993038541061823548256732313.sql
--------------------------------------------------------------------------
[DeployScripts] $ /bin/sqlplus -L ******** @/apps/jenkins-agent/workspace/HP-BIO-SNAP/DeployScripts/temp-script-16167993038541061823548256732313.sql

SQL*Plus: Release 21.0.0.0.0 - Production on Fri Mar 26 15:55:04 2021
Version 21.1.0.0.0

Copyright (c) 1982, 2020, Oracle.  All rights reserved.

ERROR:
ORA-12154: TNS:could not resolve the connect identifier specified


SP2-0751: Unable to connect to Oracle.  Exiting SQL*Plus
Process exited with status 1
--------------------------------------------------------------------------
Exit code : 1
--------------------------------------------------------------------------
java.lang.RuntimeException: Process exited with error
	at org.jenkinsci.plugins.sqlplus.script.runner.SQLPlusRunner.run(SQLPlusRunner.java:551)
	at org.jenkinsci.plugins.sqlplus.script.runner.SQLPlusRunnerBuilder.perform(SQLPlusRunnerBuilder.java:195)
	at jenkins.tasks.SimpleBuildStep.perform(SimpleBuildStep.java:123)
	at hudson.tasks.BuildStepCompatibilityLayer.perform(BuildStepCompatibilityLayer.java:80)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:804)
	at hudson.model.Build$BuildExecution.build(Build.java:197)
	at hudson.model.Build$BuildExecution.doRun(Build.java:163)
	at hudson.model.AbstractBuild$AbstractBuildExecution.run(AbstractBuild.java:514)
	at hudson.model.Run.execute(Run.java:1907)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:97)
	at hudson.model.Executor.run(Executor.java:429)
ERROR: Process exited with error
Gitcolony notification failed - java.lang.IllegalArgumentException: Invalid url: 
Finished: FAILURE
